File: backend/src/main/scala/microservices/content/objects/ResourceReviewSummaryPlannerResponse.scala
package com.typesafe.travel.content.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ResourceReviewSummaryPlannerResponse(
    resourceType: String,
    resourceId: String,
    averageRating: String,
    reviewCount: Int
)
object ResourceReviewSummaryPlannerResponse:
  given sourceEncoder: Encoder[ResourceReviewSummaryPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[ResourceReviewSummaryPlannerResponse] = deriveDecoder

File: frontend/src/microservices/content/objects/ResourceReviewSummaryPlannerResponse.ts
// ResourceReviewSummaryPlannerResponse：content 域资源评论摘要返回对象。

export type ResourceReviewSummaryPlannerResponse = {
  resourceType: string
  resourceId: string
  averageRating: string
  reviewCount: number
}
export const resourceReviewSummaryPlannerResponseFromJson = (json: string): ResourceReviewSummaryPlannerResponse =>
  JSON.parse(json) as ResourceReviewSummaryPlannerResponse
export const resourceReviewSummaryPlannerResponseToJson = (value: ResourceReviewSummaryPlannerResponse): string =>
  JSON.stringify(value)
